File: app/src/main/java/vergecurrency/vergewallet/viewmodel/StatisticsViewModel.java
package vergecurrency.vergewallet.viewmodel;


import java.util.ArrayList;
import java.util.Map;

import androidx.lifecycle.ViewModel;
import vergecurrency.vergewallet.service.model.Currency;
import vergecurrency.vergewallet.service.repository.RatesClient;

import static vergecurrency.vergewallet.service.model.PreferencesManager.getPreferredCurrency;

public class StatisticsViewModel extends ViewModel {

	private String currencyCode;
	private ArrayList<Map.Entry<String, String>> statistics ;

	public StatisticsViewModel() {
		currencyCode = Currency.getCurrencyFromJson(getPreferredCurrency()).getCurrency();
		statistics = new ArrayList<>(RatesClient.readPriceStatistics(getCurrencyCode()).entrySet());
	}

	private String getCurrencyCode() {
		return currencyCode;
	}

	public ArrayList<Map.Entry<String, String>> getStatistics() {

		return statistics;
	}


}

File: app/src/main/kotlin/vergecurrency/vergewallet/viewmodel/StatisticsViewModel.kt
package vergecurrency.vergewallet.viewmodel


import androidx.lifecycle.ViewModel
import vergecurrency.vergewallet.service.model.Currency
import vergecurrency.vergewallet.service.model.PreferencesManager
import vergecurrency.vergewallet.service.repository.RatesClient
import java.util.*

class StatisticsViewModel : ViewModel() {

    private val currencyCode: String
    val statistics: ArrayList<Map.Entry<String, String>>

    init {
        currencyCode = Currency.getCurrencyFromJson(PreferencesManager.preferredCurrency!!).currency!!
        statistics = ArrayList(RatesClient.readPriceStatistics(currencyCode).entries)
    }


}
